src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/JointGermlineCNVSegmentation.java

```diff
@@ -0,0 +1,483 @@
+            }
+            alleleNumber += samplePloidy;
+        }
+        builder.genotypes(newGenotypes);
+        if (alleleNumber > 0) {
+            if (vc.getAlternateAlleles().size() == 1) {
+                final long AC;
+                if (vc.getAlternateAllele(0).equals(GATKSVVCFConstants.DUP_ALLELE)) {
+                    AC = alleleCountMap.get(GATKSVVCFConstants.DUP_ALLELE);
+                } else {
+                    AC = alleleCountMap.get(GATKSVVCFConstants.DEL_ALLELE);
+                }
+                builder.attribute(VCFConstants.ALLELE_COUNT_KEY, AC)
+                        .attribute(VCFConstants.ALLELE_FREQUENCY_KEY, Double.valueOf(AC) / alleleNumber)
+                        .attribute(VCFConstants.ALLELE_NUMBER_KEY, alleleNumber);
+            } else {
+                final List<Long> alleleCounts = new ArrayList<>();
+                final List<Double> alleleFreqs = new ArrayList<>();
+                //if we merged and del and a dupe from different callsets, then make sure the VC has both alleles
+                //if (alleleCountMap.get(GATKSVVCFConstants.DEL_ALLELE) > 0 && alleleCountMap.get(GATKSVVCFConstants.DUP_ALLELE) > 0) {
+                //    builder.alleles(vc.getReference().getDisplayString(), GATKSVVCFConstants.DUP_ALLELE.getDisplayString(), GATKSVVCFConstants.DEL_ALLELE.getDisplayString());
+                //}
+                for (final Allele a : builder.getAlleles()) {
+                    if (a.isReference()) {
+                        continue;
+                    }
+                    alleleCounts.add(alleleCountMap.containsKey(a) ? alleleCountMap.get(a) : 0L);
+                    alleleFreqs.add(alleleCountMap.containsKey(a) ? Double.valueOf(alleleCountMap.get(a)) / alleleNumber : 0L);
+                }
+                builder.attribute(VCFConstants.ALLELE_COUNT_KEY, alleleCounts)
+                    .attribute(VCFConstants.ALLELE_FREQUENCY_KEY, alleleFreqs)
+                    .attribute(VCFConstants.ALLELE_NUMBER_KEY, alleleNumber);
+            }
+        }
+        return builder.make();
+    }
+
+    /**
+     *
+     * @param sampleName
+     * @param contig
+     * @param g may be null
+     * @return
+     */
+    private int getSamplePloidy(final String sampleName, final String contig, final Genotype g) {
+        if (!allosomalContigList.contains(contig)) {
+            return refAutosomalCopyNumber;
+        }
+        int samplePloidy = 1;
+        if (sampleDB == null || sampleDB.getSample(sampleName) == null) {
+            if (g != null) {
+                samplePloidy = g.getPloidy();
+            } else {
+                oneShotLogger.warn("Samples missing from pedigree assumed to have ploidy 1 on allosomes.");
+            }
+
+        }
+        else if (sampleDB != null && sampleDB.getSample(sampleName) != null) {
+            final Sex sampleSex = sampleDB.getSample(sampleName).getSex();
+            if (contig.equals("X") || contig.equals("chrX")) {
```

Comment:
I just came up out of a deep Wikipedia rabbit hole about Bryophytes and platyfish, but honestly the only non-human species I can imagine this being run on are canine and mouse, which both have XX/XY sex determination.  The issue at hand is that we can't trust the old `PostprocessGermlineCNVCalls` genotypes because they were all haploid.  The new ones are fixed, so they shouldn't need a pedigree file, but I was still working with old data.  Maybe this is a lot of effort for backward compatibility.  Another option is to output a schema version in the Postprocess VCF that we can parse here and then trust those genotype ploidies if it's the new version.

Reply:
I took out the allosome arg, so the tool only supports XX/XY, and I also put the schema version into the segments VCF header.  The intervals VCFs haven't been updated, but when they are I'll change the header there too.